test/docs/validation.test.js

```diff
@@ -60,7 +60,7 @@ describe('validation docs', function() {
    *
    * - All [SchemaTypes](./schematypes.html) have the built-in [required](./api.html#schematype_SchemaType-required) validator. The required validator uses the [SchemaType's `checkRequired()` function](./api.html#schematype_SchemaType-checkRequired) to determine if the value satisfies the required validator.
    * - [Numbers](./api.html#schema-number-js) have [`min` and `max`](./schematypes.html#number-validators) validators.
-   * - [Strings](./api.html#schema-string-js) have [`enum`, `match`, `minlength`, and `maxlength`](./schematypes.html#string-validators) validators.
+   * - [Strings](./api.html#schema-string-js) have [`enum`, `match`, `minlength`, `minLength`, `maxlength` and `maxLength`](./schematypes.html#string-validators) validators.
```

Comment:
I'd say these docs should just list `minlength` and `maxlength`, since listing both might imply that they do something different.

Reply:
@vkarpov15 How about we change `minlength` to `minLength`, and `maxlength` to `maxLength` everywhere in our docs so that newer people would prefer the more readable upperCase version of it, while still support the older version of it?

Reply:
How about mentioning them as `minlength`/`minLength` and `maxlength`/`maxLength`? This wouldn't imply differing functionality while being complete.

However, I'm not sure how to do the same in `SchemaStringOptions.js`, that is also breaking a test. [Duplicating the option, along with the doc comment](https://github.com/Automattic/mongoose/blob/master/lib/options/SchemaStringOptions.js#L86-L110) might become confusing for the reader.

Reply:
I'd say mentioning both can be confusing and open unnecessary questions, I'd prefer mentioning only the more readable `minLength` and `maxLength` so that newer users can default to it, why still support `minlength` and `maxlength`.

Reply:
Agreed on the readability. How about `SchemaStringOptions` and the breaking test? How should I go about it?

Reply:
We can fix that by duplicating the code definition while keeping only the docs for `minLength` and `maxLength`?
So that we replace line 86~110 https://github.com/Automattic/mongoose/blob/fcc13413acc7bf1454ffa7270587a898aec8b7d0/lib/options/SchemaStringOptions.js#L86-L110 with the following code
```js
/**
 * If set, Mongoose will add a custom validator that ensures the given
 * string's `length` is at least the given number.
 *
 * @api public
 * @property minLength
 * @memberOf SchemaStringOptions
 * @type Number
 * @instance
 */

Object.defineProperty(SchemaStringOptions.prototype, 'minLength', opts);
Object.defineProperty(SchemaStringOptions.prototype, 'minlength', opts);

/**
 * If set, Mongoose will add a custom validator that ensures the given
 * string's `length` is at most the given number.
 *
 * @api public
 * @property maxLength
 * @memberOf SchemaStringOptions
 * @type Number
 * @instance
 */

Object.defineProperty(SchemaStringOptions.prototype, 'maxLength', opts);
Object.defineProperty(SchemaStringOptions.prototype, 'maxlength', opts);
```